### DISCUSSION TITLE
How can I save plotted image as a file( .ply or .vtk)

### DISCUSSION THREAD

**Discussion by Yigitt0** [pyvista/pyvista#4014] [Q&A] [answered]
import pyvista as pv
import numpy as np

file_path=input('Enter your file path:')
points = np.loadtxt(file_path,skiprows=1 ,delimiter=";", dtype=float)
cloud = pv.PolyData(points)
cloud.plot(point_size=5,eye_dome_lighting=True)

surf2 = cloud.delaunay_2d()
#surf.plot(show_edges=True)
surf = cloud.delaunay_3d(alpha=10.8, progress_bar=True , tol=0.03, offset=0.3).extract_surface().triangulate().clean()
surf.plot()

p = pv.Plotter(notebook=False)
p.add_mesh(cloud, color='red')
p.add_mesh(surf, color=True, opacity=0.35)
p.enable_depth_peeling()
p.show()

I want to save showing "p" images as a file with the ply or vtk extension.

**Comment by banesullivan** [comment, marked as answer]
You cannot save an image as a ply or vtk file as those are mesh formats. You can save the mesh that's being plotted with the desired extension:

```py
surf.save('surf.vtk')
cloud.save('cloud.vtk')
```

**Reply by Yigitt0** [reply]
Thank you
